backend/utils/graph_db_utils_test.py

```diff
@@ -0,0 +1,64 @@
+from unittest.mock import MagicMock
+from neo4j import Driver, Session
+import pytest
+# We assign an alias to "test_connection" to avoid pytest treating it as another test function
+from utils import test_connection as verify_connection, create_goal
+
+@pytest.fixture
+def mock_session():
+    return MagicMock(spec=Session)
+
+
+@pytest.fixture
+def mock_driver(mocker, mock_session):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_driver.session.return_value = mock_session
+    return mock_driver
```

Comment:
Clever mocking this. As I understand it, the mock_session and mock_driver methods return Session and Driver MagicMock objects. And we also chain the session mock to the driver mock that then stubs any imports from utils?

Reply:
Exactly. 👍  If you pull-in the `mock_driver` into a test, then any calls to `driver.session()` will return the `mock_session`.